util.rb

```diff
@@ -0,0 +1,40 @@
+require 'sinatra/base'
+require_relative 'environment'
+
+# IAM - a resource usage metric collection and reporting system
+class Util
+  def self.max_value(data)
+    # find the max Value
+    highest = data.max { |item, item2| item[:value].abs <=> item2[:value].abs }
+    highest[:value]
+  end
+
+  def self.min_value(data)
+    # find the lowest Value
+    lowest = data.min { |item, item2| item[:value].abs <=> item2[:value].abs }
+    lowest[:value]
+  end
+
+  def self.avg_mean_value(data)
+    # find the average(Mean) of the Values
+    data.map { |item| item[:value].abs }.reduce(:+) / data.count
+  end
+
+  def self.active_bool(data)
+    # insure that Active is either True or False
+    data.each do |item|
+      return false if item[:active] != true && item[:active] != false
```

Comment:
i believe this line will always return false unless things in ruby can be neither true or false.